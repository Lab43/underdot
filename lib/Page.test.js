const Page = require('./Page');



const kant = () => new Page('testing', 'assets/kant.txt');


describe('Page class', () => {

  test('accepts source and path arguments', () => {
    const file = kant();
    expect(file.source).toBe('testing');
    expect(file.path).toBe('assets/kant.txt');
  });

  describe('parse', () => {

    test('returns the directory name', async () => {
      const file = kant();
      const { dirname } = await file.parse();
      expect(dirname).toBe('assets');
    });

    test('returns a slug', async () => {
      const file = kant();
      const { slug } = await file.parse();
      expect(slug).toBe('assets/kant');
    });

    test('returns the file extension', async () => {
      const file = kant();
      const { ext } = await file.parse();
      expect(ext).toBe('txt');
    });

    test('returns the yaml front matter', async () => {
      const file = kant();
      const { metadata } = await file.parse();
      expect(metadata).toEqual({
        'title': 'Immanuel Kant',
        'subject': 'Philosophy',
      });
    });

    test('returns the body', async () => {
      const file = kant();
      const { body } = await file.parse();
      expect(body).toMatchSnapshot();
    });

  });

});
