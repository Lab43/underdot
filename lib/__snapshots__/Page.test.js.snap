// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page class parse returns the body 1`] = `
"One who makes himself a worm cannot complain afterwards if people step on him.
â€“ Immanuel Kant
"
`;
